lib/discourse_chat_integration/provider.rb

```diff
@@ -32,6 +32,8 @@ def self.get_by_name(name)
     end
 
     def self.is_enabled(provider)
+      provider = get_by_name(provider) if provider.is_a?(String)
```

Comment:
This method expects a class, but was receiving a string from the problem check. This fixes the leak for now.